hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientAtomicLongProxy.java

```diff
@@ -140,185 +140,146 @@ public ClientAtomicLongProxy(String serviceName, String objectId) {
-                = AtomicLongCompareAndSetCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return compareAndSetAsync(expect, update).join();
     }
 
     @Override
     public long decrementAndGet() {
-        ClientMessage request = AtomicLongDecrementAndGetCodec.encodeRequest(name);
-        AtomicLongDecrementAndGetCodec.ResponseParameters resultParameters
-                = AtomicLongDecrementAndGetCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return decrementAndGetAsync().join();
     }
 
     @Override
     public long get() {
-        ClientMessage request = AtomicLongGetCodec.encodeRequest(name);
-        AtomicLongGetCodec.ResponseParameters resultParameters
-                = AtomicLongGetCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return getAsync().join();
     }
 
     @Override
     public long getAndAdd(long delta) {
-        ClientMessage request = AtomicLongGetAndAddCodec.encodeRequest(name, delta);
-        AtomicLongGetAndAddCodec.ResponseParameters resultParameters
-                = AtomicLongGetAndAddCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return getAndAddAsync(delta).join();
     }
 
     @Override
     public long getAndSet(long newValue) {
-        ClientMessage request = AtomicLongGetAndSetCodec.encodeRequest(name, newValue);
-        AtomicLongGetAndSetCodec.ResponseParameters resultParameters
-                = AtomicLongGetAndSetCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return getAndSetAsync(newValue).join();
     }
 
     @Override
     public long incrementAndGet() {
-        ClientMessage request = AtomicLongIncrementAndGetCodec.encodeRequest(name);
-        AtomicLongIncrementAndGetCodec.ResponseParameters resultParameters
-                = AtomicLongIncrementAndGetCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return incrementAndGetAsync().join();
     }
 
     @Override
     public long getAndIncrement() {
-        ClientMessage request = AtomicLongGetAndIncrementCodec.encodeRequest(name);
-        AtomicLongGetAndIncrementCodec.ResponseParameters resultParameters
-                = AtomicLongGetAndIncrementCodec.decodeResponse(invokeOnPartition(request));
-        return resultParameters.response;
+        return getAndIncrementAsync().join();
     }
 
     @Override
     public void set(long newValue) {
-        ClientMessage request = AtomicLongSetCodec.encodeRequest(name, newValue);
-        invokeOnPartition(request);
+        setAsync(newValue).join();
     }
 
     @Override
-    public ICompletableFuture<Long> addAndGetAsync(long delta) {
+    public InternalCompletableFuture<Long> addAndGetAsync(long delta) {
```

Comment:
I understand your objection because he has a covariant return type which 'exposes' a stronger interface.

In this case it is perfectly fine since it isn't visible to the end user. 

Then I also understand the 'need' for the internal method. Imho this is not needed and it leads to code noise for no good reason.